plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -130,6 +131,19 @@ public CachingHiveMetastore getMetastore()
         return metastore;
     }
 
+    @Override
+    public boolean namespaceExists(ConnectorSession session, String namespace)
```

Comment:
Do we require `ConnectorSession` parameter as it is unused?

Reply:
All `TrinoCatalog` methods take `ConnectorSession`, regardless whether this is needed. I think this is for future-proofing this, so that an implementation may choose to use that parameter for something.

However, the catalogs are created for a session, so if someone needs it, it can be passed in a constructor.

Reply:
> However, the catalogs are created for a session, so if someone needs it, it can be passed in a constructor.

I take this back. This is more complicated than it looks at the first sight.